From akpm@linux-foundation.org  Fri Apr 18 14:54:40 2008
From: Adrian Bunk <bunk@kernel.org>
Date: Fri, 18 Apr 2008 13:53:55 -0700
Subject: PCI: pci_alloc_child_bus() mustn't be __devinit
To: greg@kroah.com
Cc: linux-pci@atrey.karlin.mff.cuni.cz, akpm@linux-foundation.org, bunk@kernel.org
Message-ID: <200804182053.m3IKrt0m015127@imap1.linux-foundation.org>

From: Adrian Bunk <bunk@kernel.org>

WARNING: drivers/pci/built-in.o(.text+0xc4c): Section mismatch in reference from the function pci_add_new_bus() to the function .devinit.text:pci_alloc_child_bus()

Signed-off-by: Adrian Bunk <bunk@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/probe.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -399,8 +399,8 @@ static struct pci_bus * pci_alloc_bus(vo
 	return b;
 }
 
-static struct pci_bus * __devinit
-pci_alloc_child_bus(struct pci_bus *parent, struct pci_dev *bridge, int busnr)
+static struct pci_bus *pci_alloc_child_bus(struct pci_bus *parent,
+					   struct pci_dev *bridge, int busnr)
 {
 	struct pci_bus *child;
 	int i;
